client/src/http_default.rs

```diff
@@ -117,21 +119,21 @@ impl DefaultWebSocketRequestBuilder {
 
     /// Consumes itself to build request.
     pub fn build(self) -> Result<DefaultWebSocketStreamRequest> {
-        let mut req = self.0.and_then(|b| b.body(()).map_err(Into::into))?;
-
-        let uri = req.uri().to_string();
-        let headers = req.headers_mut();
-
-        headers.insert("Host", uri.parse()?);
-        headers.insert("Connection", "Upgrade".parse()?);
-        headers.insert("Upgrade", "websocket".parse()?);
-        headers.insert("Sec-WebSocket-Version", "13".parse()?);
-        headers.insert(
-            "Sec-WebSocket-Key",
-            tokio_tungstenite::tungstenite::handshake::client::generate_key().parse()?,
-        );
-
-        Ok(DefaultWebSocketStreamRequest(req))
+        let builder = self.0?;
+        let mut request = builder
+            .uri_ref()
+            .ok_or(eyre!("Missing URI"))?
+            .into_client_request()?;
+        if let Some(auth_header) = builder
+            .headers_ref()
+            .ok_or(eyre!("No headers found"))?
+            .get("Authorization")
+        {
+            request
+                .headers_mut()
+                .insert("Authorization", auth_header.clone());
+        }
```

Comment:
```suggestion
        for (header, value) in builder.headers().ok_or(eyre!("No headers found"))? {
            request
                .headers()
                .entry(header)
                .or_insert(value);
        }
```

I think this will be more robust. Currently you check only for one header `Authorization` but user (I mean this function user) may specify any other header as well